iam/api-client/src/main/java/com/google/iam/snippets/GetPolicy.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.iam.snippets;
+
+// [START iam_get_policy]
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.api.services.cloudresourcemanager.model.GetIamPolicyRequest;
+import com.google.api.services.cloudresourcemanager.model.Policy;
+import com.google.api.services.iam.v1.IamScopes;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+
+public class GetPolicy {
+
+  // Gets a project's policy.
+  public static Policy getPolicy(String projectId) {
+    // projectId = "my-project-id"
+
+    Policy policy = null;
+
+    CloudResourceManager service = null;
+    try {
+      service = createCloudResourceManagerService();
+    } catch (IOException | GeneralSecurityException e) {
+      System.out.println("Unable to initialize service: \n" + e.toString());
+      return policy;
```

Comment:
- Is there any more that should happen here?  
- What should they do if a policy is null?  
- Why catch at all -- it's a catastrophic error, why not just say (`throws ...`) that it's a possible result?

Reply:
I chose the throw/catch handling based on the service account snippets, which kurtisvg reviewed (https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1578). He advised avoiding throwing exceptions at the top level (see the resolved conversation here: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1578/files/ebffd067f33084556e27807dd9cbdd2f8fc386ed#diff-a832b7d097d92f54302d74c8f01b8c43).
Please lmk whether to stick to that pattern from the service account docs or change it to throw exceptions instead.

Reply:
@kurtisvg Please comment.

I'm fine if Kurtis is happy, but printing stuff out isn't really idiomatic for Java, I'm not suggesting you throw the exception, just allow it to be thrown, and document that it might be.

What you are doing is converting from Java's error system (throw/catch) to errors (like Golang).  For java users, this tends to hide bugs.  We should only catch an error if it's specific and we know what we are doing with it.  (ie handle it, do something else, etc.)

Reply:
melaniedejong@ is correct and is following the rubric guidelines. This decision is derived from the goal was that snippets be self contained to make it easier to execute. It shouldn't return any values or throw any errors. 

In an ideal world, we should be showing the user how to correctly handle or recover from an error. However, there are some situations where there isn't anything to be done (such as this one). 

IIRC the reason we went with logging unrecoverable top level errors was that if we decide to put it in the function signature, it's harder for the user to pinpoint where the error might be causing from. If we have an explicit try/catch it's more obvious what part of the code can cause that error, so if the user copies only one portion of the code they can do that.